src/librustdoc/clean/types.rs

```diff
@@ -1237,7 +1237,7 @@ impl Type {
 
     crate fn is_self_type(&self) -> bool {
         match *self {
-            Generic(ref name) => name == "Self",
+            Generic(ref name) => *name == kw::SelfUpper,
```

Comment:
```suggestion
            Generic(name) => name == kw::SelfUpper,
```

Reply:
Arf, bad artifact from the `Name` enum switch.